ui/components/component-library/button-base/button-base.types.ts

```diff
@@ -52,7 +52,7 @@ export interface ButtonBaseStyleUtilityProps
   /**
    * iconProps accepts all the props from Icon
    */
-  startIconProps?: IconProps<'span'>;
+  startIconProps?: Omit<IconProps<'span'>, 'name'>;
```

Comment:
Have to omit the name since the name is passed through the iconName prop and then all the iconProps come through here.